src/Services/FHIR/FhirLocationService.php

```diff
@@ -136,4 +143,18 @@ public function createProvenanceResource($dataRecord = array(), $encode = false)
     {
         // TODO: If Required in Future
     }
+
+    private function shouldIncludeContactInformationForLocationType($type)
+    {
+        if ($type == 'patient') {
+            // only those with access to a patient's demographic information can get their data
+            return AclMain::aclCheckCore("patients", "demo") !== false;
```

Comment:
note that when a "patient" uses the api, there is no acl mechanism. instead are forcing in the sql queries that information pertinent to the patient is available (progressed from the separate api route, to the puuid binding stuff, then to your mechanism are forcing it your queries).

Reply:
however, if this is a user that is accessing a patient demographics, then this is correct.

Reply:
So I had to put in quite a bit of effort to handle the patient situation because of how locations operate.  You'll see the updates in the code.  Essentially if you are a patient you see only facility addresses and your own patient address.